sonic-net/sonic-buildimage
# [Arista] Update phy-credo package to support 40G/100G speed change on 7800R3-48CQ2-LC

#### Why I did it
1. A recent migration of SonicV2Connector from swsssdk to swsscommon.swsscommon broke phy-credo.
2. Support more transceiver types
3. Support 40G/100G speed change on 7800R3-48CQ2-LC

##### Work item tracking
N/A

#### How I did it
1. Change the import path while keeping a fallback on the previous one for 202205.
2. Determine medium type based on transceiver compliance code.
3. Handle lane map for speed change.

#### How to verify it
This new update of phy-credo package was already merged into master branch (https://github.com/sonic-net/sonic-buildimage/commit/1302a31eee5f5284eccb13d0f16ba229cfd528bf). for 202205, I have verified that:
1. phy-credo.service no longer fails due to an import error
2. 40G/100G copper cables are handled properly.
4. links come up after speed change between 40G and 100G.

#### Which release branch to backport (provide reason below if selected)
N/A

#### Tested branch (Please provide the tested image version)





#### Description for the changelog
N/A

#### Link to config_db schema for YANG module changes
N/A

#### A picture of a cute animal (not mandatory but encouraged)
N/A
